### PR TITLE
Factor out docker operations into a wrapper client class

### PR DESCRIPTION
**Pull request by emersonknapp**
Progress toward #106 
Part of replacement for #139

Factors out docker operations into a wrapper DockerClient class that encapsulates our common usage patterns. Will be used for upcoming build stages as well.

### CODE REVIEW

File: ros_cross_compile/builders.py [hunk 1]
```diff
@@ -15,27 +15,27 @@
...
-import docker
+from ros_cross_compile.docker_client import DockerClient
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 
-def run_emulated_docker_build(image_tag: str, workspace_path: Path) -> None:
+def run_emulated_docker_build(
+    docker_client: DockerClient, image_tag: str, workspace_path: Path
```
Comment (piraka9011) [inline ros_cross_compile/builders.py]: Do we need to worry about docker_client storing some state or is it just executing Docker commands? My concern is whether docker_client will or should be immutable or not down the line.

File: ros_cross_compile/ros_cross_compile.py [hunk 2]
```diff
@@ -23,6 +23,7 @@
 from typing import List
 
 from ros_cross_compile.builders import run_emulated_docker_build
+from ros_cross_compile.docker_client import DockerClient
```
Comment (piraka9011) [inline ros_cross_compile/ros_cross_compile.py]: Does a global docker_client make sense here?

Comment (emersonknapp) [reply ros_cross_compile/ros_cross_compile.py]: global variables are an anti-pattern

File: ros_cross_compile/builders.py (hunk 1 above)
Comment (emersonknapp) [reply ros_cross_compile/builders.py]: it's a valid concern, however there is no mutable state exposed by the client. if you look at my latest pushed commit - always passing the client makes the code more testable by allowing mocking.

File: ros_cross_compile/ros_cross_compile.py (hunk 2 above)
Comment (emersonknapp) [reply ros_cross_compile/ros_cross_compile.py]: singletons are also an anti-pattern, they're just a fancy case of a global variable

File: ros_cross_compile/docker_client.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import docker
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger('Docker Client')
+
+
+class DockerClient:
+    """Simplified Docker API for this package's usage patterns."""
+
+    def __init__(self, nocache):
```
Comment (thomas-moulard) [inline ros_cross_compile/docker_client.py]: typing

Comment (thomas-moulard) [reply ros_cross_compile/docker_client.py]: `nocache` -> `disable_cache`

http://davej.io/2018/05/negative-naming.html

**Pull request merged**